agenthub/codeact_agent/__init__.py

```diff
@@ -120,5 +122,15 @@ def chat(self, message: str) -> None:
         """
         raise NotImplementedError
 
+    # TODO: implement these abstract methods
+    def add_event(self, event: Event) -> None:
+        raise NotImplementedError("Implement this abstract method")
+
+    def step(self, cmd_mgr: CommandManager) -> Event:
```

Comment:
This can be implemented by copying the body of `run` above, minus the loop (i.e. it's one step of `run`). 

I did that here: https://github.com/OpenDevin/OpenDevin/pull/109